### PR TITLE
added varnish menu style changes

### PR DESCRIPTION
**Pull request by jonborchardt**
puts us at parity with varnish menu
(also fixes some menu overrides we had for the demo site)

### CODE REVIEW

File: components/menu/IconMenuItemColumns.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import styled from 'styled-components';
+
+import Columns from '../columns';
+
+// Varnish menu Item columns for text with an icon.
+const IconMenuItemColumns = styled(Columns as any).attrs({ /* any fixes a bug, https://github.com/microsoft/TypeScript/issues/37597 */
```
Comment (codeviking) [inline components/menu/IconMenuItemColumns.tsx:6]:
```suggestion
const IconMenuItemColumns = styled(Columns as any).attrs({ /* The `any` type here fixes a bug, https://github.com/microsoft/TypeScript/issues/37597 */
```

File: components/menu/IconMenuItemColumns.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    count: 2,
+})`
+    grid-template-columns: min-content 1fr;
+    justify-items: start;
```
Comment (codeviking) [inline components/menu/IconMenuItemColumns.tsx:10]: Isn't this the default?

File: components/menu/IconMenuItemColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    align-items: center;
+    grid-gap: 0;
```
Comment (codeviking) [inline components/menu/IconMenuItemColumns.tsx:12]: Isn't `gap` the canonical version of this now? I can't recall.

File: components/menu/demo/icon-menu-item.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <Menu.Item key="3">
+      <IconMenuItemColumns>
+        <SettingOutlined />
+        <span>Option 3</span>
+      </IconMenuItemColumns>
+    </Menu.Item>
+  </Menu>,
+  mountNode,
+);
+```
```
Comment (codeviking) [inline components/menu/demo/icon-menu-item.md:44]: These new code samples are great!

**Comment by jonborchardt** [comment]
because tests take bloody forever, im going to make fixes from this pr in a new pr

**Pull request merged**